common/src/main/java/io/druid/common/utils/SerializerUtils.java

```diff
@@ -38,6 +38,34 @@
 {
   private static final Charset UTF8 = Charset.forName("UTF-8");
 
+  /**
+   * Writes the given long value into the given OutputStream, using the helperBuffer. Faster alternative to
+   * out.write(Longs.toByteArray(value)), more convenient (sometimes) than wrapping the OutputStream into {@link
+   * java.io.DataOutputStream}. The order in which the value's bytes are written to the OutputStream depends on the
+   * {@link ByteBuffer#order(ByteOrder)} of the given buffer.
+   *
+   * @param helperBuffer a heap ByteBuffer of capacity at least 8.
+   */
+  public static void writeLongToOutputStream(OutputStream out, long value, ByteBuffer helperBuffer) throws IOException
```

Comment:
Changed to separate bigEndian and nativeOrdered methods.

ByteBuffer allocation inside methods makes them pointless. Their goal is to save allocations, otherwise done by Longs.toByteArray() and Ints.toByteArray().